ui/ducks/send/send.js

```diff
@@ -2749,6 +2760,19 @@ export function signTransaction() {
         ),
       );
 
+      if (isSwapAndSend) {
+        transactionType = TransactionType.contractInteraction;
+      }
+
+      if (bestQuote?.approvalNeeded) {
+        await dispatch(
+          addTransactionAndRouteToConfirmationPage(bestQuote.approvalNeeded, {
+            sendFlowHistory: draftTransaction.history,
+            type: TransactionType.contractInteraction,
```

Comment:
Would `swapApproval` or `approve` be more accurate for this?

Reply:
`tokenMethodApprove` seems to be inferred; I never see it explicitly set anywhere. Maybe it happens automatically after contracts are deployed?

Actually now I'm thinking. Should we just submit the approval immediately then ask the user to verify the trade? Swaps doesn't seem to ask for either, but Send might need that extra layer.

Reply:
Just made it so the approval is submitted immediately and shit's kinda fire lmk what you think

Reply:
Is there someone we can ask to verify that it'll be inferred correctly?

What do you mean by submitting the approval immediately? Rn the approval + trade get queued up right?